### PR TITLE
Add description to Scylla metrics

### LINKED ISSUE #227: Copy the help text from scylla metrics onto prometheus help (visible in hover over)

### PR DESCRIPTION
**Pull request by tzach**
Fix #227 for Master dashboards

### CODE REVIEW

File: grafana/scylla-dash-per-server.master.json [hunk 1]
```diff
@@ -272,6 +272,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Counts a number of served requests",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:275]: the number

File: grafana/scylla-dash-per-server.master.json [hunk 2]
```diff
@@ -468,6 +470,7 @@
...
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:473]: the number

File: grafana/scylla-dash-per-server.master.json [hunk 3]
```diff
@@ -660,6 +664,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Number of currently pending foreground write requests",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:667]: I'm not sure that foreground and background are very clear

File: grafana/scylla-dash-per-server.master.json [hunk 4]
```diff
@@ -823,6 +829,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Number of write request failed due to a timeout",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:832]: requests

File: grafana/scylla-dash-per-server.master.json [hunk 5]
```diff
@@ -904,6 +911,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Number write requests failed due to an unavailable error",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:914]: number of

File: grafana/scylla-dash-per-server.master.json [hunk 6]
```diff
@@ -1155,6 +1165,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Number of read request failed due to a timeout",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:1168]: requests

File: grafana/scylla-dash-per-server.master.json [hunk 7]
```diff
@@ -1237,6 +1248,7 @@
                                 "step": 4
                             }
                         ],
+                        "description": "Number read requests failed due to an unavailable error",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:1251]: number of

File: grafana/scylla-dash-per-server.master.json [hunk 8]
```diff
@@ -1345,6 +1357,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Counts the total number of successful reads on this shard",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:1360]: why the difference between the reads and the writes?

File: grafana/scylla-dash-per-server.master.json [hunk 9]
```diff
@@ -1508,6 +1522,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Holds the number of currently active read operations. If this vlaue gets close to 100 we are likely to start dropping new read requests. In that case sstable_read_queue_overloads is going to get a non-zero value",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:1525]: why holds
closer not close
the description uses a metric name, it shouldn't

File: grafana/scylla-dash-per-server.master.json [hunk 10]
```diff
@@ -1589,6 +1604,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Holds the number of currently queued read operations",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:1607]: not very cleared

File: grafana/scylla-dash-per-server.master.json [hunk 11]
```diff
@@ -1751,6 +1768,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Holds a number of currently pending allocations. A non-zero value indicates that we have a bottleneck in the disk write flow",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:1771]: holds the, and I think you should remove all the holds.
Have you check that the long description looks ok?

**Comment by tzach** [comment]
Thank @amnonh good inputs.
Since the descriptions are extracted from Scylla metric help text, please open correlating issues there.

File: grafana/scylla-dash-per-server.master.json [hunk 12]
```diff
@@ -1847,6 +1865,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Counts the total number of failed read operations. Add the total_reads to this value to get the total amount of reads issued on this shard",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:1868]: don't use a metric name in a description

File: grafana/scylla-dash-per-server.master.json [hunk 13]
```diff
@@ -2009,6 +2029,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Counts a number of requests blocked due to memory pressure. A non-zero value indicates that the commitlog memory quota is not enough to serve the required amount of requests",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:2032]: the number of

File: grafana/scylla-dash-per-server.master.json [hunk 14]
```diff
@@ -2120,6 +2141,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Counts the total number of failed write operations. A sum of this value plus total_writes represents a total amount of writes attempted on this shard",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:2144]: don't use a metric name

File: grafana/scylla-dash-per-server.master.json [hunk 15]
```diff
@@ -3265,6 +3295,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Total size of memory for the cache",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:3298]: of cache memory

File: grafana/scylla-dash-per-server.master.json [hunk 16]
```diff
@@ -3380,6 +3411,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Holds a current size of allocated memory in bytes",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:3414]: the current

File: grafana/scylla-dash-per-server.master.json [hunk 17]
```diff
@@ -3461,6 +3493,7 @@
                                 "step": 1
                             }
                         ],
+                        "description": "Holds a current amount of used non-LSA memory",
```
Comment (amnonh) [inline grafana/scylla-dash-per-server.master.json:3496]: the current

**Comment by tzach** [comment]
replaced by #231

**Pull request closed without merging**